File: 2022-11/833c.py
# import math, heapq, bisect, itertools, functools
# from collections import deque, Counter, defaultdict, OrderedDict

from collections import defaultdict

def foo(n, ls):
    ans = 0
    csum = 0
    i = 0
    while i < n:
        if ls[i] == 0:
            break
        csum += ls[i]
        if csum == 0:
            ans += 1
        i += 1
        # print(ans, end=' ')
    while i < n:
        i += 1
        d = defaultdict(int)
        d[0] = 1
        csum = 0
        while i < n and ls[i] != 0:
            csum += ls[i]
            d[csum] += 1
            i += 1
        ans += max(d.values())
        # print(ans, end=' ')
    return ans

# python sample.py < input.txt

if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = int(input())
        ls = [int(i) for i in input().split()]
        print(foo(n, ls))

File: archived/2022-11/833c.cpp
#include <bits/stdc++.h>
using namespace std;

int foo(int n, vector<int>& ls) {
    int ans = 0;
    long long csum = 0;
    int cmax = 0;
    int i = 0;
    unordered_map<long long, int> d;
    while (i < n) {
        if (!ls[i]) break;
        csum += ls[i++];
        if (!csum) ans++;
    }
    while (i < n) {
        i++;
        d.clear();
        d[0] = 1;
        csum = 0;
        cmax = 1;
        while (i < n && ls[i]) {
            csum += ls[i++];
            cmax = max(cmax, ++d[csum]);
        }
        ans += cmax;
    }
    return ans;
}

// g++ sample.cpp -o run
// ./run < input.txt

int main() {
    int t, n, tmp;
    vector<int> ls;
    cin >> t;
    for (int i=0; i<t; i++) {
        cin >> n;
        ls.clear();
        ls.resize(n);
        for (int i=0; i<n; i++) {
            cin >> tmp;
            ls[i] = tmp;
        }
        cout << foo(n, ls) << endl;
    }
}